plugins/aggregators/histogram/histogram.go

```diff
@@ -151,6 +158,9 @@ func (h *HistogramAggregator) Add(in telegraf.Metric) {
 				index := sort.SearchFloat64s(buckets, value)
 				agr.histogramCollection[field][index]++
 			}
+			if h.PushOnlyOnUpdate {
+				agr.pushed = false
+			}
```

Comment:
Maybe you should name the field `updated` and set it here unconditionally to `true`. This removes the `if` here and moves the whole effect of the `h.PushOnlyOnUpdate` option down to the `Push` function. This makes the code a bit easier to follow I think.